app/src/main/java/com/example/calculator/MyFragment.kt

```diff
@@ -0,0 +1,169 @@
+                        updateText("3")
+                    }
+                    R.id.fourBTN -> {
+                        updateText("4")
+                    }
+                    R.id.fiveBTN -> {
+                        updateText("5")
+                    }
+                    R.id.sixBTN -> {
+                        updateText("6")
+                    }
+                    R.id.sevenBTN -> {
+                        updateText("7")
+                    }
+                    R.id.eightBTN -> {
+                        updateText("8")
+                    }
+                    R.id.nineBTN -> {
+                        updateText("9")
+                    }
+                    R.id.addBTN -> {
+                        updateText("+")
+                    }
+                    R.id.subtractBTN -> {
+                        updateText("-")
+                    }
+                    R.id.multiplyBTN -> {
+                        updateText("*")
+                    }
+                    R.id.divideBTN -> {
+                        updateText("/")
+                    }
+                    R.id.startBracketBTN -> {
+                        if (bracketsDifferance < 0) bracketsCorrectness = false
+                        bracketsDifferance++
+                        updateText("(")
+                    }
+                    R.id.endBracketBTN -> {
+                        if (bracketsDifferance <= 0) bracketsCorrectness = false
+                        bracketsDifferance--
+                        updateText(")")
+                    }
+                    R.id.pointBTN -> {
+                        updateText(".")
+                    }
+                    R.id.clearBTN -> {
+                        display.setText("")
+                    }
+                    R.id.equalsBTN -> {
+                        if(bracketsDifferance != 0 && !bracketsCorrectness){
+                            Toast.makeText(context,"incorrect brackets order",Toast.LENGTH_LONG)
+                                .show()
+                        }
+                        val str: String = display.text.toString()
+                        val prn = Parser().ExpressionToRPN(str)
+                        val ans = Parser().RPNToAnswer(prn)
+                        display.setText(ans.toString())
+                    }
+                    R.id.plusMinusBTN -> {
+                        val equation = BigDecimal(display.text.toString())
+                        display.setText(equation.multiply(BigDecimal(-1)).toString())
+                    }
+                    R.id.backspaceBTN -> {
+                        val length: Int = display.text.length
+                        if (length != 0) {
+                            val selection: SpannableStringBuilder =
+                                (SpannableStringBuilder(display.text))
+                            selection.replace(length - 2, length - 1, "")
+                            display.text = selection
```

Comment:
```suggestion
                            display.text = display.text.dropLast(1)
```
и не нужно создавать отдельные объекты